src/main/java/io/vertx/core/http/impl/HttpClientRequestImpl.java

```diff
@@ -672,4 +679,25 @@ private void checkResponseHandler() {
   synchronized Handler<HttpClientRequest> pushHandler() {
     return pushHandler;
   }
+
+  @Override
+  public synchronized HttpClientRequest setStreamPriority(StreamPriority streamPriority) {
+    synchronized (this) {
+      if(!streamPriority.equals(getStreamPriority())) {
+        this.streamPriority = streamPriority;
+        if(stream != null) {
+          stream.setStreamPriority(streamPriority);
+          stream.writePriorityFrame();
```

Comment:
let the stream write the frame when `setStreamPriority` is called.

Reply:
Again the VertxHttp2Stream has not know is the headers frame was written.